### DISCUSSION TITLE
What versions of microprofile specs does quarkus support?

### DISCUSSION THREAD

**Discussion by hamburml** [quarkusio/quarkus#33353] [Q&A]
Hey there,

maybe I simply did not found it but is there somewhere an overview which mp specs quarkus 3 currently follows?
There is mp config (smallrye impl), health, jwt auth, openapi, rest client ect.

We write some libraries which should be compatibel with quarkus 3 and I want to add the mp health api for example and I am wondering if I can use 4.0 or 4.0.1.

Sometimes there is a blog post like https://quarkus.io/blog/quarkus-eclipse-microprofile-3-2-compatible/ or in the changelog https://github.com/quarkusio/quarkus/releases/tag/3.0.1.Final but I would have hoped of a small table showing it. Somewhat clearer and more structured to be honest.

Thanks!

**Comment by hamburml** [comment]
I found https://quarkus.io/blog/road-to-quarkus-3/ this blog. There is Eclipse MicroProfile 6 mentioned. Does this mean all versions mentioned here https://microprofile.io/wp-content/uploads/2023/01/mricroprofile_6_release.png are supported? https://microprofile.io/2023/01/10/microprofile-6-0-release/

@gsmet What do you say? Doesn't it make sense to document these compatibilities somewhere?

**Comment by terletimehm** [comment]
I am also looking for this info. It's hard to find info about compatibility with Java EE and MicroProfile. There are some hints in blog posts, but it's not very easy to aggregate version specific info from the blog.

Is Quarkus compliant with Java EE? It would be good with info about what parts are supported and what parts are not.
What version of MicroProfile does Quarkus 2.16 support? And what about 3.0? Like @hamburml says, some kind of table would be nice.

**Reply by hamburml** [reply]
Quarkus 3.0 supports MicroProfile 6.0 which consists of https://microprofile.io/wp-content/uploads/2023/01/mricroprofile_6_release.png. Java/Jakarta EE and MicroProfile can work together. Quarkus 2 I am not sure but I would skim through the dependencies till I find somewhere these dependencies https://mvnrepository.com/artifact/org.eclipse.microprofile

**Comment by sberyozkin** [comment]
Hey @radcortez Would it make sense to create some summary re MP ? It is MP JWT 2.1 in case mentioning specific spec versions would be on the map :-)

**Reply by radcortez** [reply]
Yes, we can create such page.

**Reply by hamburml** [reply]
Ping - just want to remember :)

**Reply by radcortez** [reply]
A proposal:

https://github.com/quarkusio/quarkusio.github.io/pull/1743

**Comment by radcortez** [comment]
By the way, the source of truth for supported specifications is always the BOM:
https://github.com/quarkusio/quarkus/blob/main/bom/application/pom.xml

**Reply by terletimehm** [reply]
I'm guessing not every jakarta api dependency in the bom means that the spec is supported? For example these are in the bom, but I believe they are not fully supported.

- Jakarta Mail
- Jakarta Authentication
- Jakarta Authorization

Maybe you're saying that if you know a spec is supported you can go to the bom to get the version. That's probably true.

**Reply by hamburml** [reply]
That's how I've always done it so far. However, for the enterprise context, it is easier with an "official page/document" than with an XML file.

**Reply by radcortez** [reply]
> Maybe you're saying that if you know a spec is supported you can go to the bom to get the version. That's probably true.

Correct!
